python/lsst/summit/testing/analysis/m1m3/inertia_compensation_system.py

```diff
@@ -0,0 +1,464 @@
+#!/usr/bin/python
+import logging
+from datetime import timedelta
+
+import numpy as np
+import pandas as pd
+from astropy import units as u
+from astropy.time import Time
+from lsst.summit.utils.efdUtils import getEfdData
+from lsst.summit.utils.tmaUtils import TMAEvent, TMAEventMaker
+from plots import inertia_compensation_system
+
+# Configure the logger
+formatter = logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s")
+formatter.datefmt = "%Y-%m-%d %H:%M:%S"
+
+handler = logging.StreamHandler()
+handler.setLevel(logging.ERROR)
+handler.setFormatter(formatter)
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+logger.addHandler(handler)
+
+# TMAEventMaker needs to be instantiated only once.
+event_maker = TMAEventMaker()
```

Comment:
Ohhhh, now I see how my comment below about the undefined local variable is wrong. But yeah, this is way too illegal, even for me. You can't have code which executes like this on import, that's deeply, _deeply_ evil. Happy to help you sort this out if you need a hand architecting around this though, just drop me a DM.

Reply:
Moved to inside the `if __name__ == "__main__"`.